sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -706,6 +696,161 @@ Operation<BatchImageGenerationOperationResponse> imagesOperation
+        /// Transcription request.
+        /// Requesting format 'json' will result on only the 'text' field being set.
+        /// For more output data use 'verbose_json.
+        /// </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="deploymentId"/> or <paramref name="audioTranscriptionOptions"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="deploymentId"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual Response<AudioTranscription> GetAudioTranscription(string deploymentId, AudioTranscriptionOptions audioTranscriptionOptions, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
+            Argument.AssertNotNull(audioTranscriptionOptions, nameof(audioTranscriptionOptions));
+
+            using var scope = ClientDiagnostics.CreateScope("OpenAIClient.GetAudioTranscription");
+            scope.Start();
+
+            audioTranscriptionOptions.InternalNonAzureModelName = deploymentId;
+
+            RequestContent content = audioTranscriptionOptions.ToRequestContent();
+            RequestContext context = FromCancellationToken(cancellationToken);
+            Response rawResponse = default;
+
+            try
+            {
+                using HttpMessage message = CreateGetAudioTranscriptionRequest(deploymentId, content, context);
+                rawResponse = _pipeline.ProcessMessage(message, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+
+            return Response.FromValue(AudioTranscription.FromResponse(rawResponse), rawResponse);
+        }
+
+        /// <summary> Transcribes and translates input audio into English text. </summary>
+        /// <param name="deploymentId"> Specifies either the model deployment name (when using Azure OpenAI) or model name (when using non-Azure OpenAI) to use for this request. </param>
+        /// <param name="audioTranslationOptions">
+        /// Translation request.
+        /// Requesting format 'json' will result on only the 'text' field being set.
+        /// For more output data use 'verbose_json.
+        /// </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="deploymentId"/> or <paramref name="audioTranslationOptions"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="deploymentId"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual async Task<Response<AudioTranscription>> GetAudioTranslationAsync(string deploymentId, AudioTranslationOptions audioTranslationOptions, CancellationToken cancellationToken = default)
```

Comment:
The return type is `AudioTranscription` when it should be `AudioTranslation`.

Reply:
Egads, yes -- thank you for catching this. Another one of the big problems with having so much custom code.